examples/bigbench_hard/bigbench_hard.py

```diff
@@ -0,0 +1,361 @@
+            elif method == "cot_zeroshot":
+                full_prompt = self.cot_zeroshot_prompt.format(instruction=f"{self.sys_prompt}\n{prompt}",
+                                                              input=input_str)
+            else:
+                raise ValueError(f"generate_prompt: Unknown method: {method}")
+
+        logging.info("full prompt: %s", full_prompt)
+        return full_prompt
+
+    def aggregation_prompt(self, state_dicts: List[Dict], **kwargs) -> str:
+        pass
+
+    def improve_prompt(self, **kwargs) -> str:
+        pass
+
+    def score_prompt(self, state_dicts: List[Dict], **kwargs) -> str:
+        pass
+
+    def validation_prompt(self, **kwargs) -> str:
+        pass
+
+
+class BigBenchHardParser(parser.Parser):
+    """
+    BigBenchHardParser provides the parsing of the language model responses to the BigBench-Hard Dataset.
+
+    Inherits from the Parser class and implements its abstract methods.
+    """
+
+    def parse_generate_answer(self, state: Dict, texts: List[str]) -> List[Dict]:
+        """
+        Parse the response from the language model for a generate prompt.
+
+        :param state: The thought state used to generate the prompt.
+        :type state: Dict
+        :param texts: The responses to the prompt from the language model.
+        :type texts: List[str]
+        :return: The new thought states after parsing the response from the language model.
+        :rtype: List[Dict]
+        """
+        new_states = []
+        for text in texts:
+            if state["method"].startswith("io") or state["method"].startswith("cot"):
+                answer_str = extract_answer(text)
+                if answer_str is None:
+                    logging.warning(
+                        f"Could not parse step answer: {text}. Returning None."
+                    )
+                new_state = state.copy()
+                new_state["current"] = answer_str
+                new_state["phase"] = 2
+                new_states.append(new_state)
+            else:
+                raise ValueError(f"parse_generate_answer: Unknown method: {state['method']}")
+        return new_states
+
+    def parse_aggregation_answer(self, response: str, **kwargs) -> Union[Dict, List[Dict]]:
+        pass
+
+    def parse_improve_answer(self, response: str, **kwargs) -> Dict:
+        pass
+
+    def parse_validation_answer(self, response: str, **kwargs) -> bool:
+        pass
+
+    def parse_score_answer(self, response: str, **kwargs) -> List[float]:
+        pass
+
+
+def io() -> operations.GraphOfOperations:
+    """
+    Generates the Graph of Operations for the IO method.
+
+    :return: Graph of Operations
+    :rtype: GraphOfOperations
+    """
+    operations_graph = operations.GraphOfOperations()
+
+    operations_graph.append_operation(operations.Generate(1, 1))
+    operations_graph.append_operation(operations.GroundTruth(test_answer))
```

Comment:
I think for plotting, you have to add a final scoring method.. we should discuss this